### PR TITLE
Feature/email verification rate limit

### PR DESCRIPTION
**Pull request by jace-garcia**

### CODE REVIEW

File: apps/badgeuser/models.py [hunk 1]
```diff
@@ -35,6 +35,16 @@ class Meta:
         verbose_name = _("email address")
         verbose_name_plural = _("email addresses")
 
+    def generate_verification_time_cache_key(self):
+        return str(super(CachedEmailAddress, self).email) + "_verification_request_date"
```
Comment (coffindragger) [inline apps/badgeuser/models.py]: could we use something a little clearer here? like:

    "{}_verification_request_date".format(self.email)

File: apps/badgeuser/api_v1.py [hunk 2]
```diff
@@ -109,7 +113,28 @@ def put(self, request, id, **kwargs):
...
+                    time_delta = current_time - last_request_time
+                    if time_delta > RATE_LIMIT_DELTA:
+                        send_confirmation = True
+
+                if send_confirmation:
+                    email_address.send_confirmation(request=request)
+                    email_address.set_last_verification_sent_time(datetime.datetime.now())
+                else:
+                    remaining_time_obj = RATE_LIMIT_DELTA - (datetime.datetime.now() - last_request_time)
+                    remaining_time_rep = str((remaining_time_obj.seconds//60)%60) + " minutes and " \
```
Comment (coffindragger) [inline apps/badgeuser/api_v1.py]: I'd prefer to see string.format used here as well instead of manually calling str() and using + concatenation

Comment (jace-garcia) [reply apps/badgeuser/api_v1.py]: fixed

File: apps/badgeuser/models.py (hunk 1 above)
Comment (jace-garcia) [reply apps/badgeuser/models.py]: fixed

**Pull request merged**